src/tests/D3MHub.t.sol

```diff
@@ -16,704 +16,647 @@
 
 pragma solidity ^0.8.14;
 
-import {DssTest} from "dss-test/DssTest.sol";
-import "./interfaces/interfaces.sol";
+import "dss-test/DssTest.sol";
+import "dss-interfaces/Interfaces.sol";
 
-import {D3MHub} from "../D3MHub.sol";
-import {D3MOracle} from "../D3MOracle.sol";
-import "../pools/ID3MPool.sol";
-import "../plans/ID3MPlan.sol";
+import { D3MHub } from "../D3MHub.sol";
+import { D3MOracle } from "../D3MOracle.sol";
+import { ID3MPool } from "../pools/ID3MPool.sol";
+import { ID3MPlan } from "../plans/ID3MPlan.sol";
 
-import {D3MTestPool} from "./stubs/D3MTestPool.sol";
-import {D3MTestPlan} from "./stubs/D3MTestPlan.sol";
-import {D3MTestGem} from "./stubs/D3MTestGem.sol";
-import {D3MTestRewards} from "./stubs/D3MTestRewards.sol";
+import { TokenMock } from "./mocks/TokenMock.sol";
 
-interface Hevm {
-    function warp(uint256) external;
+contract PoolMock is ID3MPool {
 
-    function store(
-        address,
-        bytes32,
-        bytes32
-    ) external;
+    TokenMock public dai;
+    TokenMock public gem;
+
+    bool public preDebt;
+    bool public postDebt;
+    uint256 public maxDeposit;
+
+    constructor(address _vat, address _hub, address _dai, address _gem) {
+        dai = TokenMock(_dai);
+        gem = TokenMock(_gem);
+        maxDeposit = type(uint256).max;
+        VatAbstract(_vat).hope(_hub);
+    }
+
+    function deposit(uint256) external override {
+    }
+
+    function withdraw(uint256 wad) external override {
+        dai.transfer(msg.sender, wad);
+    }
+
+    function exit(address dst, uint256 wad) external override {
+        gem.transfer(dst, wad);
+    }
+
+    function quit(address dst) external override {
+        gem.transfer(dst, gem.balanceOf(address(this)));
+    }
+
+    function preDebtChange() external override {
+        preDebt = true;
+    }
+
+    function postDebtChange() external override {
+        postDebt = true;
+    }
+
+    function resetPrePostDebt() external {
+        preDebt = false;
+        postDebt = false;
+    }
+
+    function assetBalance() public view returns (uint256) {
+        return dai.balanceOf(address(this)) + gem.balanceOf(address(this));
+    }
+
+    function setMaxDeposit(uint256 _maxDeposit) external {
+        maxDeposit = _maxDeposit;
+    }
+
+    function maxWithdraw() external view override returns (uint256) {
+        return dai.balanceOf(address(this));
+    }
+
+    function redeemable() external view returns (address) {
+        return address(gem);
+    }
+
+    function decreaseIdleLiquidity(uint256 amount) external {
```

Comment:
Can I get a TLDR of what exactly means this increasing/decreasing `DAI` supply against the `gem`? I'm a bit confused why it is being used this way.
Is it to represent how much is lent? 
If that's the case I'm a bit confused why when it is used says this "liquidity before simulating other user's withdraw", shouldn't be there term "borrow" instead of "withdraw"?

Reply:
Yeah so the fundamental thing is that the pool has some notion of "idle liquidity" which is DAI that is immediately available. The mock is just using a `gem` to represent some IOU asset for the DAI so the proper `assetBalance()` can be calculated. Really this could have probably been further simplified to just a number in the mock instead of an ERC20.

For the second point borrow or withdraw is fine. Both result in the DAI sitting in the pool going away. In the borrow case another user has borrowed some of the previously idle DAI. In the withdraw case someone was previously lending and withdrew. Of course this is all in the context of Aave/Compound and shouldn't probably have that language in the comments. I'll remove the comment.